src/data_hub/_api.py

```diff
@@ -182,18 +183,38 @@ def dynamic_function(
                 title="Deep Origin error: [Name conflicts]",
             )
 
-        if debug:
-            print(f"Response request = {response.http_request}")
-            print("Response headers:")
-            _print_dict(dict(response.headers), json=False)
-            print(f"Response = {response.http_response}")
+        if not isinstance(response, dict):
             response = response.json()
 
-        if hasattr(response, "data"):
-            return response.data
+        if "data" in response.keys():
+            response = response["data"]
+            if isinstance(response, list):
+                response = [Box(item) for item in response]
+            else:
+                response = Box(response)
         else:
-            # this should never happen
-            return response
+            response = Box(response)
+
+        if _stash:
+            print("Stashing response")
+            stash_loc = (
+                Path(__file__).resolve().parent.parent.parent
+                / "tests"
+                / "fixtures"
+                / "responses"
+                / f"{method_path}.json"
+            )
+            stashed_data = PersistentDict(stash_loc)
+            key = json.dumps(kwargs, sort_keys=True)
+
+            if isinstance(response, list):
+                stashed_response = response[:10]
+            else:
+                stashed_response = response
+
+            stashed_data[key] = stashed_response
```

Comment:
if _stash, then we save the responses in a JSON file on disk. this will then be consumed by the MockClient, which allows us to exactly reproduce the state of a live instance using a local mock, without needing any HTTP traffic